buttons.h

```diff
@@ -0,0 +1,170 @@
+        string _color;
+
+    
+        // Getters:
+        auto id() const -> const int& { return _id; }
+        auto pic() const -> const bitmap& { return _pic; }
+        auto btn() const -> const sprite& { return _btn; }
+        auto location() const -> const point_2d& { return _location; }
+        auto x() const -> const int& { return _x; }
+        auto y() const -> const int& { return _y; }
+        auto centre_x() const -> const int& { return _centre_x; }
+        auto centre_y() const -> const int& { return _centre_y; }
+        auto color() const -> const string& { return _color; }
+
+        // Enumeration of button options
+        enum Color
+        {
+            PLAY,
+            EXIT,
+            OPTIONS
+        };
+
+        // Default Constructor
+        Button(){}
+
+        // First Overloaded Constructor
+        Button(Color c, float scale = 1)
+        {
+            // button color
+            this->_color = btn_color(c);
+            // load button color image
+            this->_pic = load_bitmap(to_string(c), this->_color);
+            // create sprite from image
+            this->_btn = create_sprite(this->_pic);
+            // scale the sprite
+            sprite_set_scale(this->_btn, scale);
+        }
+        
+        // Second Overloaded Constructor
+        // Calculates the position in the window
+        Button(Color c, float x, float y, float scale = 1)
+        {
+            // button color
+            this->_color = btn_color(c);
+            // load button color image
+            this->_pic = load_bitmap(to_string(c), this->_color);
+            // create sprite from image
+            this->_btn = create_sprite(this->_pic);
+            // get the centre points of the sprite 
+            this->_centre_x = sprite_width(this->_btn) /2;
+            this->_centre_y = sprite_height(this->_btn)/2;
+            // store the intended location 
+            this->_x = x * X_CELL; _y = y * Y_CELL;
+            // set button sprite centre point to intended location
+            sprite_set_x(this->_btn, this->_x - this->_centre_x);
+            sprite_set_y(this->_btn, this->_y - this->_centre_y);
+            // store the window centre point of button as location
+            this->_location = center_point(this->_btn);
+            // scale the sprite
+            sprite_set_scale(this->_btn, scale);
+        }
+
+        ~Button(){}
+
+        virtual void draw_button() = 0;
+        virtual void action() = 0;
+
+        void draw_screen_guides()
+        {
+            draw_circle(COLOR_BLACK, WIDTH*0.5 , HEIGHT*0.5 , 3);
+            draw_circle(COLOR_BLACK, WIDTH*0.25, HEIGHT*0.5 , 3);
+            draw_circle(COLOR_BLACK, WIDTH*0.75, HEIGHT*0.5 , 3);
+            draw_circle(COLOR_BLACK, WIDTH*0.5 , HEIGHT*0.25, 3);
+            draw_circle(COLOR_BLACK, WIDTH*0.5 , HEIGHT*0.75, 3);                                    
+        }
+
+        string btn_color(Color c)
```

Comment:
Maybe this function should exist within one of the sub-classes, considering we are going to have a number of button assets I assume for menu, options, etc.